modules/login/php/requestaccount.class.inc

```diff
@@ -79,10 +83,36 @@ class RequestAccount extends \NDB_Form
      * @param array $values The values that were submitted to the page.
      *
      * @return array a (possibly empty) array of errors
+     *
+     * @see https://github.com/google/recaptcha
      */
     function _validate($values): array
     {
         $errors = array();
+        // Verify reCAPTCHA
+        $err = array();
+        $reCAPTCHAPrivate = \NDB_Factory::singleton()
+            ->config()
+            ->getSetting('reCAPTCHAPrivate') ?? '';
+
+        // Only validate reCAPTCHA if a private key exists.
+        if (!empty($reCAPTCHAPrivate)) {
+            $success = false;
+            $resp    = (new \ReCaptcha\ReCaptcha($reCAPTCHAPrivate))
+                ->verify(
+                    $values['g-recaptcha-response'] ?? '',
+                    $_SERVER['REMOTE_ADDR']
+                );
+            $success = $resp->isSuccess();
+            // FIXME This leverages the error handling for the "site" text field.
```

Comment:
I couldn't figure out how to get an error code to show up under the reCAPTCHA part so this is a bit hack-y but is functioning. I'm open to suggestions

Reply:
@johnsaigle maybe `$resp` returns null if error? I'm wondering what happens when an error is thrown.

Reply:
Oh sorry I didn't add the new file to Git. 😅